catalog/core/catalog-core-api/src/main/java/ddf/catalog/CatalogFramework.java

```diff
@@ -323,7 +323,10 @@ ResourceResponse getResource(ResourceRequest request, String resourceSiteName)
   /**
    * Returns information for each {@link ddf.catalog.source.Source} that is endpoint-addressable in
    * {@link CatalogFramework}, including its own {@link ddf.catalog.source.CatalogProvider}, based
-   * on the contents of the {@link SourceInfoRequest}.
+   * on the contents of the {@link SourceInfoRequest}. Implementations of this method should
```

Comment:
❓Instead of saying _Implementations of this method should periodically check the availability of each Source_, should we say that the call is non-blocking and that the availability status may be based on an earlier check or something along those lines?

Reply:
```java
/**
 * Returns information for each {@link ddf.catalog.source.Source} that is endpoint-addressable in
 * {@link CatalogFramework}, including its own {@link ddf.catalog.source.CatalogProvider}, based
 * on the contents of the {@link SourceInfoRequest}. This method is a non-blocking alternative to
 * {@link ddf.catalog.source.Source#isAvailable()}. The availability of {@link
 * ddf.catalog.source.Source}s returned are based on previous availability checks by the {@link
 * CatalogFramework}.
 *
 * <p>The {@link SourceInfoRequest} specifies either:
 *
 * <ol>
 *   <li>an enterprise source search, which includes the catalog provider and all federated and
 *       connected sources
 *   <li>a list of requested source IDs, which are all federated sources
 *   <li>the catalog provider only
 * </ol>
 *
 * @param sourceInfoRequest the {@link SourceInfoRequest}
 * @return {@link SourceInfoResponse}
 * @throws SourceUnavailableException if the source indicated in the {@link SourceInfoRequest} is
 *     <code>null</code>, not found, or cannot provide the requested information
 */
SourceInfoResponse getSourceInfo(SourceInfoRequest sourceInfoRequest)
    throws SourceUnavailableException;
```